tests/FixerConfiguration/FixerConfigurationResolverTest.php

```diff
@@ -0,0 +1,166 @@
+        $configuration->resolve(array());
+    }
+
+    public function testResolveWithDefault()
+    {
+        $option = new FixerOption('foo', 'Bar.');
+        $option->setDefault('baz');
+        $configuration = new FixerConfigurationResolver(array(
+            $option,
+        ));
+
+        $this->assertSame(
+            array('foo' => 'baz'),
+            $configuration->resolve(array())
+        );
+    }
+
+    public function testResolveWithAllowedTypes()
+    {
+        $option = new FixerOption('foo', 'Bar.');
+        $option->setAllowedTypes(array('int'));
+        $configuration = new FixerConfigurationResolver(array(
+            $option,
+        ));
+
+        $this->assertSame(
+            array('foo' => 1),
+            $configuration->resolve(array('foo' => 1))
+        );
+
+        $this->setExpectedException('Symfony\Component\OptionsResolver\Exception\InvalidOptionsException');
+        $this->assertSame(
+            array('foo' => 1),
+            $configuration->resolve(array('foo' => '1'))
+        );
+    }
+
+    public function testResolveWithAllowedValues()
+    {
+        $option = new FixerOption('foo', 'Bar.');
+        $option->setAllowedValues(array(true, false));
+        $configuration = new FixerConfigurationResolver(array(
+            $option,
+        ));
+
+        $this->assertSame(
+            array('foo' => true),
+            $configuration->resolve(array('foo' => true))
+        );
+
+        $this->setExpectedException('Symfony\Component\OptionsResolver\Exception\InvalidOptionsException');
+        $this->assertSame(
+            array('foo' => 1),
+            $configuration->resolve(array('foo' => 1))
+        );
+    }
+
+    public function testResolveWithUndefinedOption()
+    {
+        $configuration = new FixerConfigurationResolver(array(
+            new FixerOption('bar', 'Bar.'),
+        ));
+
+        $this->setExpectedException('Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException');
+        $configuration->resolve(array('foo' => 'foooo'));
+    }
+
+    public function testResolveWithNormalizers()
+    {
+        $option = new FixerOption('foo', 'Bar.');
+        $option->setNormalizer(function (Options $options, $value) {
+            return (int) $value;
+        });
+        $configuration = new FixerConfigurationResolver(array(
+            $option,
+        ));
+
+        $this->assertSame(
+            array('foo' => 1),
+            $configuration->resolve(array('foo' => '1'))
+        );
+
+        $exception = new InvalidOptionsException('');
+        $option->setNormalizer(function (Options $options, $value) use ($exception) {
+            throw $exception;
+        });
+
+        $catched = null;
+        try {
+            $configuration->resolve(array('foo' => '1'));
+        } catch (InvalidOptionsException $catched) {
+        }
+
+        $this->assertSame($exception, $catched);
```

Comment:
Then I think the test is over engineered.  Aren't you testing OptionResolver of Symfony here ?